godot-core/src/builtin/color_hsv.rs

```diff
@@ -0,0 +1,247 @@
+///
+/// ## Examples
+///
+/// ```
+/// use godot::builtin::{Color, ColorHsv};
+///
+/// // ColorHsv can be constructed either from Hue value in 0.0 - 1.0 representation, or degree
+/// // representation 0.0 - 360.0.
+/// let c_hsv = ColorHsv::from_hsv(0.74, 0.69, 0.18);
+/// let mut c_hsv_deg = ColorHsv::from_hsv_deg(266.4, 0.69, 0.18);
+///
+/// assert_eq!(c_hsv, c_hsv_deg);
+///
+/// // If any value would be set to values below minimal or above maximal viable value, it will
+/// // be clamped silently.
+/// let c_hsv_min_max = ColorHsv::from_hsv(1.0, 0.0, 1.0);
+///
+/// c_hsv_deg.set_h(100.);
+/// c_hsv_deg.set_s(-10.);
+/// c_hsv_deg.set_v(1.12);
+///
+/// assert_eq!(c_hsv_min_max, c_hsv_deg);
+///
+/// // Two way convertion: Color -> ColorHsv -> Color is not entirely lossless. Such repeatable
+/// // convertions should be avoided, as they could build up to significant values if values
+/// // are mutated both on `Color` and `ColorHsv`.
+/// let color1 = Color::from_rgb(0.74, 0.69, 0.18);
+/// let color2: Color = ColorHsv::from(color1).into();
+///
+/// let diff_r = (color1.r - color2.r).abs();
+/// let diff_g = (color1.g - color2.g).abs();
+/// let diff_b = (color1.b - color2.b).abs();
+///
+/// assert_ne!(color1, color2);
+///
+/// assert!(diff_r < 0.001);
+/// assert!(diff_g < 0.001);
+/// assert!(diff_b < 0.001);
+/// ```
+///  
+/// ## Reference
+/// - Smith, Alvy Ray. "Color gamut transform pairs." ACM Siggraph Computer Graphics 12.3 (1978): 12-19.
+#[derive(Debug, Clone, Copy, PartialOrd, Default)]
+pub struct ColorHsv {
+    h: f32,
+    s: f32,
+    v: f32,
+    a: f32,
+}
+
+// Manual implementation because Hue minimal value (0.) is equal to maximal value (1.)
+impl PartialEq for ColorHsv {
+    fn eq(&self, other: &Self) -> bool {
+        (self.h == other.h || (self.h - other.h).abs() == 1.)
+            && self.s == other.s
+            && self.v == other.v
+            && self.a == other.a
+    }
+}
+
+impl ColorHsv {
+    /// Construct from Hue, Saturation, Value where all values are between `0.0` and `1.0`.
+    ///
+    /// If values outside of that range are provided, they will be clamped to valid range. To add `alpha` value, use  `with_alpha()` method.
+    pub fn from_hsv(h: f32, s: f32, v: f32) -> Self {
+        ColorHsv {
+            h: h.clamp(0., 1.),
+            s: s.clamp(0., 1.),
+            v: v.clamp(0., 1.),
+            a: 1.,
+        }
+    }
+
+    /// Construct from Hue, Saturation, Value where Hue is represented in deegrees (`0.0` - `360.0`).
+    ///
+    /// Saturation, Value and Alpha have to be provided in range between `0.0` an `1.0`. If values outside of valid ranges are provided,
+    /// they will be clamped to a valid range. To add `alpha` value, use `with_alpha()` method.
+    pub fn from_hsv_deg(h: f32, s: f32, v: f32) -> Self {
+        ColorHsv {
+            h: h.clamp(0., 360.) / 360.,
```

Comment:
I'm not sure if `clamp` is the most intuitive behavior here -- if you have an angle, wouldn't you rather wrap it? (i.e. modulo 360°)

Reply:
Great suggestion - not only there but for another constructor and both setters also. It would allow easily rotating around hue circle and possibly also remove the issue with hue `PartialEq`